src/js/directives/autoHideShow.js

```diff
@@ -0,0 +1,147 @@
+            this.meister.container.removeEventListener('touchmove', this.onMouseMove);
+            this.meister.container.removeEventListener('mouseleave', this.onMouseLeave);
+
+            this.showControls();
+            this.showCursor();
+        } else {
+            this.meister.container.addEventListener('mousedown', this.onMouseDown);
+            this.meister.container.addEventListener('touchstart', this.onMouseDown);
+
+            this.meister.container.addEventListener('mousemove', this.onMouseMove);
+            this.meister.container.addEventListener('touchmove', this.onMouseMove);
+            this.meister.container.addEventListener('mouseleave', this.onMouseLeave);
+
+            // Start hide countdown.
+            this.mouseTimeout = setTimeout(this.hideCursor.bind(this), TIME_TILL_FADE);
+            this.controlsTimeout = setTimeout(this.hideControls.bind(this), TIME_TILL_FADE);
+        }
+    }
+
+    /**
+     * Callback for hideCursor-Event, triggers a Meister-event
+     * @return {null}
+     */
+    hideCursor() {
+        if (this.dragging) {
+            return;
+        }
+        this.meister.trigger('uiEvent:hideCursor', {});
+    }
+
+    /**
+     * Callback for showCursor-Event, triggers a Meister-event
+     * @return {null}
+     */
+    showCursor() {
+        this.meister.trigger('uiEvent:showCursor', {});
+    }
+
+    /**
+     * Callback for hideControls-Event, triggers a Meister-event
+     * @return {null}
+     */
+    hideControls() {
+        if (this.dragging || !this.meister.playing) {
+            return;
+        }
+        this.meister.trigger('uiEvent:hideControls', {});
+    }
+
+    /**
+     * Callback for showControls-Event, triggers a Meister-event
+     * @return {null}
+     */
+    showControls() {
+        this.meister.trigger('uiEvent:showControls', {});
+    }
+
+    /* Events */
+    onMouseDown() {
+        this.dragging = true;
+        window.addEventListener('mouseup', this.onMouseUp);
+        window.addEventListener('touchend', this.onMouseUp);
+    }
+
+    onMouseUp() {
+        this.dragging = false;
+        // Fake mouse movement
+        this.onMouseMove();
+
+        window.removeEventListener('mouseup', this.onMouseUp);
+        window.removeEventListener('touchend', this.onMouseUp);
+    }
+
+    onMouseMove() {
+        clearTimeout(this.mouseTimeout);
+        clearTimeout(this.controlsTimeout);
+
+        this.showControls();
+        this.showCursor();
+
+        this.mouseTimeout = setTimeout(this.hideCursor.bind(this), TIME_TILL_FADE);
+        this.controlsTimeout = setTimeout(this.hideControls.bind(this), TIME_TILL_FADE);
+    }
+
+    onMouseLeave() {
+        clearTimeout(this.mouseTimeout);
+        clearTimeout(this.controlsTimeout);
+
+        this.hideControls();
+        this.showCursor();
+    }
+
+
+}
+
+const autoHideShow = (meister, config) => {
+    new AutoHideShowControls(meister, config);
```

Comment:
Good point. 
I'd suggest a abstract class for directives, in which an `unload` method is defined. `autoHideShow` should be an instance of that abstract and implement it's own `unload`, which unbinds events and cleans up.
`HtmlUi` will keep references to directive in memory to trigger the `unload` in directives when the `protoPlugin.unload` method is called.